osmoutils/slice_helper_test.go

```diff
@@ -91,3 +92,134 @@ func TestContainsDuplicateDeepEqual(t *testing.T) {
 		require.Equal(t, tt.want, got)
 	}
 }
+
+func TestIThSmallest(t *testing.T) {
```

Comment:
shall we add float and string test cases to cover one of each of `Integer | Float | ~string` in `constraints.Ordered`?